src/scripts/ci_build.py

```diff
@@ -684,7 +688,10 @@ def main(args=None):
             if options.compiler_cache is not None:
                 cmds.append([options.compiler_cache, '--show-stats'])
 
-            make_targets = ['libs', 'tests', 'cli']
+            make_targets = ['libs', 'cli']
+
+            if run_test_command is not None:
+                make_targets += ['tests']
```

Comment:
Is that a good idea? Basically not to build the test executable if this particular CI run isn't planning to run it anyway?

Also: This selection of make targets is sort-of a repetition of the selection of build targets in the helper method in this very same script. Perhaps we should consider cleaning that up at some point:

https://github.com/randombit/botan/blob/68ca91208e2c640fab835f3ba40d5e37145beac4/src/scripts/ci_build.py#L89-L111

Reply:
Hrm - that will disable building the tests also on CI jobs where we don't currently run the tests (Android NDK, iOS, Emscripten) which isn't great. Considering this just saves on building the tests in a nightly job, I'd be inclined to continue building the tests here.

I agree this script is getting a little out of control :sweat_smile: I had ideas on rebuilding it such that most of this decision logic is driven by a text configuration rather than embedding everything about the targets into the script itself, which might help.

Reply:
Reverted.